test/built-ins/Map/iterable-calls-set.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 23.1.1.1
+description: >
+  new Map calls `set` for each item on the iterable argument in order.
+info: >
+  Map ( [ iterable ] )
+
+  ...
+  9. Repeat
+    ...
+    k. Let status be Call(adder, map, «k.[[value]], v.[[value]]»).
+  ...
+includes: [compareArray.js]
+---*/
+
+var mapSet = Map.prototype.set;
+var counter = 0;
+
+var iterable = [["foo", 1], ["bar", 2]];
+var results = [];
+
+Map.prototype.set = function(k, v) {
+  counter++;
+  results.push([k, v]);
+  mapSet.call(this, k, v);
+};
+
+var s = new Map(iterable);
+
+assert.sameValue(counter, 2, "`Map.prototype.set` called twice.");
+
+assert(compareArray(results[0], iterable[0]));
+assert(compareArray(results[1], iterable[1]));
```

Comment:
This would be a good place to assert the context of the `set` method when invoked by the Map constructor